app/views/admin/configs/show.html.erb

```diff
@@ -234,11 +228,52 @@
                                              placeholder: Constants::SiteConfig::DETAILS[:recaptcha_secret_key][:placeholder] %>
                           </div>
                         </div>
+                        <div class="mt-6">
+                          <% if SiteConfig.allow_both_email_signup_and_login %>
+                            <button
+                              class="crayons-btn crayons-btn--danger <%= cannot_disable_email_auth ? "crayons-tooltip" : "" %>"
+                              data-action="click->config#activateEmailAuthModal"
+                              data-tooltip="<%= cannot_disable_email_auth ? disable_email_auth_tooltip_text : "" %>"
+                              <%= cannot_disable_email_auth ? "disabled" : "" %>>
```

Comment:
Could do something similar here, too:
```ruby
def disable_button_class
  (SiteConfig.invite_only_mode || authentication_enabled_providers.none?) ? "disabled" : ""
end
```

```suggestion
                              <%= disabled_button_class %>>
```

If chose to do this, then ultimately, you wouldn't need the `cannot_disable_email_auth` helper, because you'd have two new, more specific helpers determining the class for you.